scripts/ast-for-funcs/ast-for-funcs.scala

```diff
@@ -0,0 +1,27 @@
+import scala.collection.JavaConverters._
+
+import io.circe.syntax._
+import io.circe.{Encoder, Json}
+import io.shiftleft.codepropertygraph.generated.nodes.AstNode
+
+import org.apache.tinkerpop.gremlin.structure.VertexProperty
+
+implicit val encodeVertex: Encoder[AstNode] = (node: AstNode) =>
+  Json.obj(
+    ("id", Json.fromString(node.toString)),
+    ("properties", Json.fromValues(node.properties().asScala.toList.map { p: VertexProperty[_] =>
+      Json.obj(
+        ("key", Json.fromString(p.key())),
+        ("value", Json.fromString(p.value().toString))
+      )
+    }))
+  )
+
+Json.obj(
+  ("file", Json.fromString(cpg.file.name.l.head)),
+  ("functions", Json.fromValues(cpg.method.name.l.map { methodName =>
+    val method = cpg.method.name(methodName)
+    Json.obj(("function", Json.fromString(methodName)),
+      ("AST", Json.fromValues(method.astChildren.l.map(_.asJson))))
+  }))
+)
```

Comment:
I think it'd be nicer to define a case class for this, rather than writing it by hand:
```scala
import io.circe.generic.semiauto._

case class AstForFuncsFunction(function: String, AST: List[AstNode])
case class AstForFuncsResult(file: String, functions: List[AstForFuncsFunction])

implicit val encodeVertex: Encoder[AstNode] = ... // As before
implicit val encodeFuncResult: Encoder[AstForFuncsResult] = deriveEncoder
implicit val encodeFuncFunction: Encoder[AstForFuncsFunction] = deriveEncoder

AstForFuncResult(
  cpg.file.name.l.head,
  cpg.method.name.l.map { methodName =>
    val method = cpg.method.name(methodName)
    AstForFuncsFunction(method, method.astChildren.l)
  }
).asJson
```

Reply:
Done. Thanks :+1: